ui/src/app/cohort-review/detail-tab-table/detail-tab-table.component.tsx

```diff
@@ -0,0 +1,290 @@
+  columns: Array<any>;
+  domain: string;
+  filterType: PageFilterType;
+  participantId: number;
+  workspace: WorkspaceData;
+}
+
+export interface DetailTabTableState {
+  data: Array<any>;
+  loading: boolean;
+  start: number;
+  rows: number;
+  sortField: string;
+  sortOrder: number;
+}
+
+export const DetailTabTable = withCurrentWorkspace()(
+  class extends React.Component<DetailTabTableProps, DetailTabTableState> {
+    dt: any;
+    constructor(props: DetailTabTableProps) {
+      super(props);
+      this.state = {
+        data: null,
+        loading: true,
+        start: 0,
+        rows: 25,
+        sortField: null,
+        sortOrder: 1
+      };
+    }
+
+    componentDidMount() {
+      this.getParticipantData();
+    }
+
+    componentDidUpdate(prevProps: any) {
+      if (prevProps.participantId !== this.props.participantId) {
+        this.setState({
+          data: null,
+          loading: true,
+        });
+        this.getParticipantData();
+      }
+    }
+
+    getParticipantData() {
+      const {cdrVersionId, id, namespace} = this.props.workspace;
+      const {cid} = urlParamsStore.getValue();
+      const pageFilterRequest = {
+        page: 0,
+        pageSize: 10000,
+        sortOrder: SortOrder.Asc,
+        sortColumn: this.props.columns[0].name,
+        pageFilterType: this.props.filterType,
+        domain: this.props.domain,
+      } as PageFilterRequest;
+
+      cohortReviewApi().getParticipantData(
+        namespace,
+        id,
+        +cid,
+        +cdrVersionId,
+        this.props.participantId,
+        pageFilterRequest
+      ).then(response => {
+        response.items.forEach(item => {
+          if (this.props.domain === DomainType[DomainType.VITAL]
+            || this.props.domain === DomainType[DomainType.LAB]) {
+            item['itemTime'] = moment(item.itemDate, 'YYYY-MM-DD HH:mm Z').format('hh:mm a z');
+          }
+          item.itemDate = moment(item.itemDate).format('YYYY-MM-DD');
+        });
+        this.setState({
+          data: response.items,
+          loading: false,
+        });
+      });
+    }
+
+    onSort = (event: any) => {
+      this.setState({sortField: event.sortField, sortOrder: event.sortOrder});
+    }
+
+    columnSort = (sortField: string) => {
+      if (this.state.sortField === sortField) {
+        const sortOrder = this.state.sortOrder === 1 ? -1 : 1;
+        this.setState({sortOrder});
+      } else {
+        this.setState({sortField, sortOrder: 1});
+      }
+    }
+
+    onPage = (event: any) => {
+      this.setState({start: event.first});
+    }
+
+    render() {
+      const {data, loading, rows, start, sortField, sortOrder} = this.state;
+      let pageReportTemplate;
+      if (data !== null) {
+        const lastRowOfPage = (start + rows) > data.length
+          ? start + rows - (start + rows - data.length) : start + rows;
+        pageReportTemplate = (start + 1) + ' - ' + lastRowOfPage + ' of ' + data.length
```

Comment:
nit: Using backtick strings with interpolation would probably make this easier to read.